lib/Cake/Utility/File.php

```diff
@@ -89,7 +89,7 @@ public function __construct($path, $create = false, $mode = 0755) {
 			$this->name = basename($path);
 		}
 		$this->pwd();
-		$create && !$this->exists() && $this->safe($path) && $this->create();
+		$create && !$this->exists() && $this->safe($path) && $this->create($mode);
```

Comment:
The `$create` param too is used for the directory as well as file. I agree awfullness of this API. We should at least make it sensible in 3.0.